### PR TITLE
Attempt to fix output stage

### PR DESCRIPTION
**Pull request by korczis**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/310/ : FAILURE in 7s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/245/ : FAILURE in 1h 13m 20s
- https://ci.intgdc.com/job/gooddata-lcm-zuul/37/ : SUCCESS in 2m 30s (non-voting)

File: lib/gooddata/models/automated_data_distribution.rb [hunk 1]
```diff
@@ -6,7 +6,19 @@
...
-    attr_accessor :output_stage
+    attr_writer :output_stage
+
+    def output_stage
+      if @output_stage
+        @output_stage
+      else
+        url = project.uri.gsub('/gdc/projects/', '/gdc/dataload/projects/') + '/outputStage'
+        data = project.client.get(url)
+        # TODO: We need to pass ADS but how to get it?
```
Comment (vhtien) [inline lib/gooddata/models/automated_data_distribution.rb]: I don't understand. What is the context? In outputStage, the only way we can get ADS is that outputStage is configured its schemas.

**Comment by vhtien** [comment]
I think this PR is covered in #794

**Comment by korczis** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/316/ : FAILURE in 11s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/251/ : SUCCESS in 43m 04s
- https://ci.intgdc.com/job/gooddata-lcm-zuul/43/ : SUCCESS in 2m 35s (non-voting)

File: lib/gooddata/models/automated_data_distribution.rb [hunk 2]
```diff
@@ -8,6 +8,18 @@ module GoodData
   class AutomatedDataDistribution < Rest::Resource
     attr_writer :output_stage
 
+    def output_stage
```
Comment (vhtien) [inline lib/gooddata/models/automated_data_distribution.rb]: do not agree with this version. The old one in https://github.com/gooddata/gooddata-ruby/pull/791/files#diff-dad39358c99f5e64258fd991e1752949R33 is better.

Comment (korczis) [inline lib/gooddata/models/automated_data_distribution.rb]: ok

File: lib/gooddata/models/automated_data_distribution.rb [hunk 3]
```diff
@@ -19,15 +25,7 @@ def process
...
-      self.output_stage = GoodData::AdsOutputStage.create({ ads: ads, project: project, client: project.client }.merge(opts))
-      output_stage
-    end
-
-    def output_stage
-      return @output_stage if @output_stage
-
-      @output_stage = GoodData::AdsOutputStage[project: project, client: project.client]
-      @output_stage
+      @output_stage = GoodData::AdsOutputStage.create({ads: ads, project: project, client: project.client}.merge(opts))
```
Comment (panjan) [inline lib/gooddata/models/automated_data_distribution.rb]: We should use `project.add_output_stage` here.

Comment (vhtien) [inline lib/gooddata/models/automated_data_distribution.rb]: it's the same @panjan. We're already in `project.add` class

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/318/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/253/ : SUCCESS in 43m 28s
- https://ci.intgdc.com/job/gooddata-lcm-zuul/45/ : FAILURE in 2m 47s (non-voting)

**Comment by vhtien** [comment]
looks like we have to add certificate for jruby

**Pull request merged**